scipy/special/orthogonal.py

```diff
@@ -194,21 +194,27 @@ def _gen_roots_and_weights(n, mu0, an_func, bn_func, f, df, symmetrize, mu):
     dy /= np.abs(dy).max()
     w = 1.0 / (fm * dy)
 
+    if log_mu0 is not None:
+        if symmetrize: raise ValueError('cannot set symmetrize=True with log_mu0 not None')
+        
+        log_w = np.log(w) + log_mu0 - np.log(w.sum())
+        
+        if mu: return x, log_w, log_mu0
+        return x, log_w
+
     if symmetrize:
         w = (w + w[::-1]) / 2
         x = (x - x[::-1]) / 2
 
     w *= mu0 / w.sum()
 
-    if mu:
-        return x, w, mu0
-    else:
-        return x, w
+    if mu: return x, w, mu0
+    return x, w
 
 # Jacobi Polynomials 1               P^(alpha,beta)_n(x)
 
 
-def roots_jacobi(n, alpha, beta, mu=False):
+def roots_jacobi(n, alpha, beta, mu=False, log_values=False):
```

Comment:
This option should be tested.

Reply:
added test here (didn't push it yet), although I get the following failure in test_orthogonal.py#L331:

when log_weights=True, I'm getting an difference of .023, which is a relative difference of about 1.69e-15 (mu is of the scale of 10^13). So the test is a bit to harsh for taking exponents and logs. what should I do about that?

regarding the normal behavior (log_weights=False), all tests pass.

Reply:
see my comment in the discussion